compiler/rustc_symbol_mangling/src/legacy.rs

```diff
@@ -55,6 +55,26 @@ pub(super) fn mangle(
 
     let hash = get_symbol_hash(tcx, instance, instance_ty, instantiating_crate);
 
+    if let ty::InstanceDef::DropGlue(_drop_in_place, ty) = instance.def {
+        // Use `{{drop}}::<$TYPE>::$hash` as name for the drop glue instead of
+        // `core::mem::drop_in_place::$hash`.
```

Comment:
cc @eddyb does this seem right to you?

Reply:
Why not just enable type parameters (on the original path), if you want them on the legacy mangling?
All you have to do is use `instance.substs` instead of `&[]` below (conditionally on `matches!(instance.def, ty::InstanceDef::DropGlue(..)`, I'm guessing, to avoid legacy symbol size blow-up?).

Or you can use `RUSTFLAGS=-Zsymbol-mangling-version=v0` for now instead. I'm still a bit bummed that whole thing took forever to get upstreamed, and I haven't checked its progress since that happened either.

Reply:
Done